drivers/sensor/adxl375/adxl375.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2023 Aaron Chan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl375
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include "zephyr/sys/byteorder.h"
+#include <zephyr/sys/__assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "adxl375.h"
+
+LOG_MODULE_REGISTER(ADXL375, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl375_check_id(const struct device *dev)
+{
+	struct adxl375_data *data = dev->data;
+	uint8_t device_id = 0;
+
+	int ret = data->hw_tf->read_reg(dev, ADXL375_DEVID, &device_id);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (ADXL375_DEVID_VAL != device_id) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int adxl375_set_odr_and_lp(const struct device *dev, uint32_t data_rate,
+				  const bool low_power)
+{
+	struct adxl375_data *data = dev->data;
+
+	if (low_power) {
+		data_rate |= 0x8;
+	}
+
+	return data->hw_tf->write_reg(dev, ADXL375_BW_RATE, data_rate);
+}
+
+static int adxl375_set_op_mode(const struct device *dev, enum adxl375_op_mode op_mode)
+{
+	struct adxl375_data *data = dev->data;
+
+	return data->hw_tf->write_reg(dev, ADXL375_POWER_CTL, op_mode);
+}
+
+static int adxl375_set_data_format(const struct device *dev, uint8_t val)
+{
+	struct adxl375_data *data = dev->data;
+
+	return data->hw_tf->write_reg(dev, ADXL375_DATA_FORMAT, val);
+}
+
+static int adxl375_init(const struct device *dev)
+{
+	int ret;
+	const struct adxl375_dev_config *cfg = dev->config;
+
+	ret = cfg->bus_init(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize sensor bus.");
+		return ret;
+	}
+
+	ret = adxl375_check_id(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to get valid device ID.");
+		return ret;
+	}
+
+	ret = adxl375_set_odr_and_lp(dev, cfg->odr, cfg->lp);
+	if (ret < 0) {
+		LOG_ERR("Failed to set ODR and LP mode");
+		return ret;
+	}
+
+	ret = adxl375_set_data_format(dev, ADXL375_DATA_FORMAT_DEFAULT_BITS);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize data format");
+		return ret;
+	}
+
+	ret = adxl375_set_op_mode(dev, ADXL375_MEASUREMENT);
+	if (ret < 0) {
+		LOG_ERR("Failed to put in standby mode");
+	}
+
+	return ret;
```

Comment:
if you check throughout the function `ret != 0` then you can return 0 directly here.